src/lib/github.ts

```diff
@@ -21,6 +22,116 @@ export class Github {
         return hide;
     }
 
+    /**
+    * TODO: Incomplete
+    *
+    * Returns a repo's branches, and the commits on those branches.
+    * Note that branches share identical commits. They are non-unique.
+    * 
+    * @param {string} url
+    */
+
+    public static async getBranches(url: string): Promise<IBranch[]> {
+        const urlL = url.slice(19);
+        let branches: IBranch[] = [];
+        await fetch(global_APIURLs.github + urlL + "/branches", {
+            headers: new Headers(this.headers_githubRequestAuth)
```

Comment:
Following the review above, this needs to be changed to `headers: this.headers_githubRequestAuth`